File: amazia-core/src/main/java/com/example/CreateProduct/controller/CreateProductController.java
package com.example.CreateProduct.controller;

import com.example.CreateProduct.service.CreateProductService;
import com.example.shared.entity.Product;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api")
public class CreateProductController {

    private final CreateProductService createProductService;

    public CreateProductController(CreateProductService createProductService) {
        this.createProductService = createProductService;
    }

    @PostMapping("/products")
    public ResponseEntity<Product> create(@Valid @RequestBody Product product) {
        return ResponseEntity.status(HttpStatus.CREATED).body(createProductService.create(product));
    }
}

File: amazia-console/app/Product/Controller/CreateProductController.php
<?php

namespace App\Product\Controller;

use App\Http\Controllers\Controller;
use App\Product\Service\CreateProductService;
use Illuminate\Http\Request;

class CreateProductController extends Controller
{
    private CreateProductService $service;

    public function __construct(CreateProductService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $response = $this->service->create($request->all());
        return response()->json($response->json(), $response->status());
    }
}
